tutorials/01_03_geometry.ipynb

```diff
@@ -0,0 +1,404 @@
+   "metadata": {},
+   "source": [
+    "## 3d plot (k3d)\n",
+    "\n",
+    "In the previous section we learned how to generate simple 3d-plots from a groove and a seam length using the `plot` method.\n",
+    "By default, `plot` relies on matplotlib as rendering backend.\n",
+    "We have already mentioned before that this is not ideal for 3d renderings.\n",
+    "Therefore, the `weldx` package offers an alternative for jupyter notebook users.\n",
+    "You can select the `k3d` package as rendering backend as follows:\n",
+    "\n",
+    "> HINT: Interactive 3d plots are not visible in the plain html documentation. \n",
+    "        Click on the binder link button at the beginning of the tutorial to start a jupyter-lab session of this tutorial."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Uncomment next line to create a k3d plot\n",
+    "# geometry.plot(backend=\"k3d\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we got a nice 3d rendering of the geometry with a closed surface that we shift and turn as we like.\n",
+    "However, because `k3d` uses equal axis scaling, we notice that the specimen appears to be rather slim.\n",
+    "The reason for this is that the provided groove types only specify the shape of the groove and not the actual specimen width.\n",
+    "But we can provide one during creation of the geometry using the `width` parameter.\n",
+    "Note that this only works with groove types and not with a plain `Profile`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> TODO Remove `Q_` below"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "geometry_2 = Geometry(groove, t, width=Q_(\"20cm\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now lets plot the geometry again:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Uncomment next line to create a k3d plot\n",
+    "# geometry_2.plot(profile_raster_width=\"5mm\", trace_raster_width=\"16cm\", backend=\"k3d\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Export 3d geometry into a CAD file\n",
+    "\n",
+    "The `Geometry` class also features a `to_file` method.\n",
+    "It allows you to export the 3d data to different file formats like `.ply` or `.stl`.\n",
+    "Using this function, you can inspect the geometry in a CAD program or model viewer of your choice:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "geometry.to_file(\n",
+    "    \"weldx_geometry.stl\", profile_raster_width=Q_(\"2mm\"), trace_raster_width=Q_(\"50mm\")\n",
+    ")"
+   ]
```

Comment:
I get the following `RuntimeWarning` in the last cell (STL export):
```python
...\site-packages\meshio\stl\_stl.py:185: RuntimeWarning: invalid value encountered in true_divide
  normals = (normals.T / nrm).T
```
Is this expected?

Reply:
Not nice, but expected. This error simply says that we have one or more degenerated triangles (no surface cause all points lie on a straight line) in the mesh. They are located at the top and the bottom surfaces and result from the simplified triangulation algorithm we are using for those surfaces. They have no visual impact so you won't see it in the plots. I have removed the warning in the tutorial by simply changing the rasterization parameters but a real fix would require us to use a proper triangulation algorithm or to filter the degenerated triangles.

Reply:
ok, if it is as simple as choosing better rasterization parameters then that should work 👍